app/services/recurring_contribution/pagarme_subscription.rb

```diff
@@ -0,0 +1,53 @@
+class RecurringContribution::PagarmeSubscription
+  def initialize(plan_id, user, payment_method, credit_card)
+    @credit_card = credit_card
+    @payment_method = normalize_payment_method(payment_method)
+    @plan_id = plan_id
+    @user = user
+  end
+
+  def build
+    RecurringContribution::PagarmeAPI::Subscription.create(attributes)
+  end
+
+  private
+
+  def attributes
+    if credit_card?
+      default_attributes.merge(credit_card_attributes)
+    else
+      default_attributes
+    end
+  end
+
+  def default_attributes
+    {
+      plan: RecurringContribution::PagarmeAPI::Plan.find(@plan_id),
+      payment_method: @payment_method,
+      postback_url: "http://test.com/postback",
+      customer: { email: @user.email }
+    }
+  end
+
+  def credit_card_attributes
```

Comment:
since you haven't changed the key names, you can slice the `@credit_card` hash to return only the entries that you want:
```ruby
@credit_card.slice(:card_number, :card_holder_name, :card_expiration_month, :card_expiration_year, :card_cvv)
```